lang/cs/Org.Apache.REEF.Common/Exceptions/RecoverableNetworkException.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Org.Apache.REEF.Common.Exceptions
+{
+    /// <summary>
+    /// Encapsulates <see cref="Exception#ToString"/> for an Exception related to 
+    /// the Network that can be recovered from.
+    /// </summary>
+    [Serializable]
+    public sealed class RecoverableNetworkException : Exception
```

Comment:
I called this `RecoverableNetworkException` rather than `RecoverableIMRUNetworkingError` because it's not strictly related to `IMRU` -- currently, we only look for this in the IMRU framework, but theoretically, we could do so in other frameworks too.